### PR TITLE
Terrain & imagery provider ready deprecation fixes

### PR DESCRIPTION
**Pull request by ggetz**
* Fixes https://github.com/CesiumGS/cesium/issues/11274 
    * This is resolved by ensuring the existence of either `_ready` or `ready` before checking the value.
    * The logic is a bit more complicated in these checks as we don't want to check `ready` unless completely necessary. Otherwise an unneeded deprecation warning is thrown when accessing `ready` for our build-in providers.
* Fixes deprecation warning typo reported in https://github.com/CesiumGS/cesium/issues/11195#issuecomment-1555937605
* Fixes https://github.com/CesiumGS/cesium/issues/11303#issuecomment-1558379153

For testing, here's a [local sandcastle with custom terrain and imagery providers](http://localhost:8080/Apps/Sandcastle/index.html#c=5Vdpb9s2GP4rhD/Jg0vHWXMscYJ5rtsGcJeiyZZuMGDQ0mubK0VqJOXUG/zfx0O3lcXZ9m0BAkPk897HI4WCK402FB5BoivE4RGNQdE0xj+7s2DWCd3zWHBNKAc56/TQnzOO0IIomJItyItc5CYmK5Bbd4iXUsQPQrIoOw2cEEJKbxmUIoJXQXf2Eo8mn25G0/nDzf37+XT0w2R617Oyu26vZvcjDb9Y60vCFJirXfdyxkMXELP3ygTkI8MqBA6YVvxTBpujF4yEXz4QuWBgRJ4K5qMUGxqBHKktDwPrSRlDBs4hDj9SCvRNFHx7Pjjuzrh1rmIIE5asiTF3hE8aNwtJV2vNQdkAjvGRufbxYBJFQQXZdTEsUx5qKjgap0qLuOEKCrooy7xeU4XnmjLKV3fhGmKoV/wdiJUkyZqG9xVMXjcJoSZ8Vandp/zEhfsGVhJABR5t/16dneCjXvl8fF57fHV6Vr/+Dp+d5I/d7IKn8QLk7XIKG2C/ghTGNVCfL9DgbwG/lABgjCZK0Kjwe5Kf4Id3d+evfWu5VOY5AimFnGyA63qG3FFgsTsLnvHbxW8mCTiCpRkNk/IEpKYmCa2lwIkUWuhtAtkESSDR9iIvzwr0ha3W1TVKudcYoX4fCYm0TMG52fNmM1GjOKYKDtGQQbEEJdgGAquxW1dZbYy6yqLDbC8Z0zqVvKWbLtGu6WTRMv9AHS7E9xQbGDzQSK/bQkcng+N9+BrsUB2Gj8lXGqexa6k2iaMGnPKXwO1/2WAHp6YU2UvHmqiR3SbjNeG83Yeig7I16QWfaVNTgd9TUNoBzCSU3n3toW0PMRtxD2WoYs9kbtf3aKGdCeK3fcBTZqRnHYz7bjOrvheZT8VKzMUGpFl7OOGrWce7vDvYcRO53QNjCRHVquZ6081iUl5oIjHs8xaI0QFVAwfoz2knyVd0tmPaDAYlp3ketrmpL6UqUQW5zm6dNErZVs64N81lyH2PMw7dhv+WWw7f0bkllq/7D35Wjf4YtKThxDpbsngjMtsYE6VNu2mICsqoC78VcvTebYuYJMFT1IkLnzOeOX3dexJrrP6Y8dRnx08j7YwHR24HZ4TyHJ00Y/m/0Ml/vjUztn8mq/n2swXLOmTbtkn8dD5aOjLXg9PL8tCTTnmaudbMXvYmU05L0X6Zc2+IJvmsmBfgdLm0771W4CfK9fnIgLaB9+CbzGi3eLN69ETpforDnA79b3a8K7vxoASZ5Ls2fmoGy1RZvW5kzXtw3qyVKj0zzX0UDNBw6DWU7h3koK2eTd9oQygj/g1/r4Qtq/qwDFgya1gwHay3LzJS+0bRjVVc5YaGH271dnqdofukus5L+z2NEyE1SiULDLVqiBNm1p3qL1LzvaRxqJTfpAgN+1XRYUQ3iEZXLZ98KGREKXOzNJx9R/+AWed62Df4PVGbETPCt569LWw9uJ76Q4zxsG8e2yW1EGxBZEPzXw). Sandcastles related to terrain or imagery should also be verified.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ggetz!
* :heavy_check_mark: Signed CLA found.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: packages/widgets/Source/Viewer/Viewer.js [hunk 1]
```diff
@@ -2153,7 +2153,13 @@ function zoomToOrFly(that, zoomTarget, options, isFlight) {
 
       if (defined(zoomTarget.imageryProvider)) {
         // This is here for backward compatibility. It can be removed when readyPromise is removed.
-        rectanglePromise = zoomTarget.imageryProvider._readyPromise.then(() => {
+        let promise = Promise.resolve();
+        if (defined(zoomTarget.imageryProvider)) {
+          promise = zoomTarget.imageryProvider._readyPromise;
+        } else if (defined(zoomTarget.imageryProvider.readyPromise)) {
```
Comment (jjhembd) [inline packages/widgets/Source/Viewer/Viewer.js:2159]: The `else if` condition will never be met, if `zoomTarget.imageryProvider` is not defined.
Should the `if` condition instead be `defined(zoomTarget.imageryProvider._readyPromise)`?

**Review by jjhembd — COMMENTED** [review]
Thanks @ggetz, this looks good overall. I had one question

File: packages/widgets/Source/Viewer/Viewer.js (hunk 1 above)
Comment (ggetz) [reply packages/widgets/Source/Viewer/Viewer.js:2159]: Good catch @jjhembd, that's an error on my part. It should now be fixed.

**Comment by jjhembd** [comment]
Looks good @ggetz! Merging now

**Pull request merged**